lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -47,13 +61,30 @@ def update_aws_object(volume)
   end
 
   def destroy_aws_object(volume)
-    converge_by "delete EBS volume #{new_resource.name} in #{region}" do
-      volume.delete
+    status = volume.status
+    case status
+    when :in_use
+      attachment = volume.attachments.first
+      detach(volume, :instance => attachment.instance, :device => attachment.device)
+      delete(volume)
+    else
+      delete(volume)
     end
   end
 
   private
 
+  def desired_instance
```

Comment:
I understand the intent.  I was referring to the definition of the word: "a strong feeling of wanting to have something or wishing for something to happen".  The word implies an emotional context, and made me laugh of bit.  I was looking for a more objective word to describe the intent :)